components/org.wso2.carbon.identity.organization.user.invitation.management/src/main/java/org/wso2/carbon/identity/organization/user/invitation/management/dao/UserInvitationDAOImpl.java

```
@@ -232,6 +236,26 @@ public Invitation getInvitationByConfirmationCode(String confirmationCode)
         return invitation;
     }
 
+    @Override
+    public Invitation getInvitationWithRolesByConfirmationCode(String confirmationCode)
+            throws UserInvitationMgtServerException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+            PreparedStatement getInvitationIdFromCodePrepStat =
+                    connection.prepareStatement(GET_INVITATION_ID_FROM_CONFIRMATION_CODE)) {
+            getInvitationIdFromCodePrepStat.setString(1, confirmationCode);
+            try (ResultSet resultSet = getInvitationIdFromCodePrepStat.executeQuery()) {
+                if (resultSet.next()) {
+                    String invitationId = resultSet.getString(COLUMN_NAME_INVITATION_ID);
+                    return getInvitationByInvitationId(invitationId);
+                }
+                return null;
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```

Do we need to throw the runtime exception here?